Development/nmos/node_server.h

```diff
@@ -76,9 +90,9 @@ namespace nmos
         // and the experimental Logging API and Settings API, according to the specified data models and callbacks
         nmos::server make_node_server(nmos::node_model& node_model, nmos::experimental::node_implementation node_implementation, nmos::experimental::log_model& log_model, slog::base_gate& gate);
 
-        nmos::server make_node_server(nmos::node_model& node_model, nmos::transport_file_parser parse_transport_file, nmos::details::connection_resource_patch_validator validate_merged, nmos::connection_resource_auto_resolver resolve_auto, nmos::connection_sender_transportfile_setter set_transportfile, nmos::connection_activation_handler connection_activated, nmos::experimental::log_model& log_model, slog::base_gate& gate);
-        nmos::server make_node_server(nmos::node_model& node_model, nmos::transport_file_parser parse_transport_file, nmos::connection_resource_auto_resolver resolve_auto, nmos::connection_sender_transportfile_setter set_transportfile, nmos::connection_activation_handler connection_activated, nmos::experimental::log_model& log_model, slog::base_gate& gate);
-        nmos::server make_node_server(nmos::node_model& node_model, nmos::connection_resource_auto_resolver resolve_auto, nmos::connection_sender_transportfile_setter set_transportfile, nmos::connection_activation_handler connection_activated, nmos::experimental::log_model& log_model, slog::base_gate& gate);
+        nmos::server make_node_server(nmos::node_model& node_model, nmos::load_cert_handler load_rsa, nmos::load_cert_handler load_ecdsa, nmos::load_dh_param_handler load_dh_param, nmos::load_cacert_handler load_cacert, nmos::transport_file_parser parse_transport_file, nmos::details::connection_resource_patch_validator validate_merged, nmos::connection_resource_auto_resolver resolve_auto, nmos::connection_sender_transportfile_setter set_transportfile, nmos::connection_activation_handler connection_activated, nmos::experimental::log_model& log_model, slog::base_gate& gate);
+        nmos::server make_node_server(nmos::node_model& node_model, nmos::load_cert_handler load_rsa, nmos::load_cert_handler load_ecdsa, nmos::load_dh_param_handler load_dh_param, nmos::load_cacert_handler load_cacert, nmos::transport_file_parser parse_transport_file, nmos::connection_resource_auto_resolver resolve_auto, nmos::connection_sender_transportfile_setter set_transportfile, nmos::connection_activation_handler connection_activated, nmos::experimental::log_model& log_model, slog::base_gate& gate);
+        nmos::server make_node_server(nmos::node_model& node_model, nmos::load_cert_handler load_rsa, nmos::load_cert_handler load_ecdsa, nmos::load_dh_param_handler load_dh_param, nmos::load_cacert_handler load_cacert, nmos::connection_resource_auto_resolver resolve_auto, nmos::connection_sender_transportfile_setter set_transportfile, nmos::connection_activation_handler connection_activated, nmos::experimental::log_model& log_model, slog::base_gate& gate);
```

Comment:
I think these ought to be left alone for backward compatibility? And there's probably no need to add more overloads as the example apps demonstrate the preferred "fluent initialization" approach.

```suggestion
        nmos::server make_node_server(nmos::node_model& node_model, nmos::transport_file_parser parse_transport_file, nmos::details::connection_resource_patch_validator validate_merged, nmos::connection_resource_auto_resolver resolve_auto, nmos::connection_sender_transportfile_setter set_transportfile, nmos::connection_activation_handler connection_activated, nmos::experimental::log_model& log_model, slog::base_gate& gate);
        nmos::server make_node_server(nmos::node_model& node_model, nmos::transport_file_parser parse_transport_file, nmos::connection_resource_auto_resolver resolve_auto, nmos::connection_sender_transportfile_setter set_transportfile, nmos::connection_activation_handler connection_activated, nmos::experimental::log_model& log_model, slog::base_gate& gate);
        nmos::server make_node_server(nmos::node_model& node_model, nmos::connection_resource_auto_resolver resolve_auto, nmos::connection_sender_transportfile_setter set_transportfile, nmos::connection_activation_handler connection_activated, nmos::experimental::log_model& log_model, slog::base_gate& gate);
```

And fix up the **.cpp** file too.

Reply:
> I think these ought to be left alone for backward compatibility? And there's probably no need to add more overloads as the example apps demonstrate the preferred "fluent initialization" approach.
> 
> And fix up the **.cpp** file too.

So if leaving those make_node_server() functions as before (without overload), then all those functions are just for backward compatibility.  Am I correct that if users want to use the new handlers, they must use 
`nmos::server make_node_server(nmos::node_model& node_model, nmos::experimental::node_implementation node_implementation, nmos::experimental::log_model& log_model, slog::base_gate& gate);`?

Reply:
Exactly - as the **nmos-cpp-node** example does on master already:
https://github.com/sony/nmos-cpp/blob/9577343d53fd31d5e05e5e183f818fa5904f8492/Development/nmos-cpp-node/main.cpp#L100

Reply:
okay, should we add a comment for those deprecated make_node_server functions?